src/app/FakeLib/FSIHelper.fs

```diff
@@ -353,16 +353,28 @@ let internal runFAKEScriptWithFsiArgsAndRedirectMessages printDetails (FsiArgs(f
                             File.Delete("FSI-ASSEMBLY.dll.mdb")
 
                         let dynamicAssemblies = 
-                            System.AppDomain.CurrentDomain.GetAssemblies() 
+                            System.AppDomain.CurrentDomain.GetAssemblies()
                             |> Seq.filter(fun assem -> assem.IsDynamic)
-                            |> Seq.filter(fun assem -> assem.GetName().Name <> "FSI-ASSEMBLY")
+                            |> Seq.map(fun assem -> assem.GetName().Name)
+                            |> Seq.filter(fun assem -> assem <> "FSI-ASSEMBLY")
+                            // General Reflection.Emit helper (most likely harmless to ignore)
+                            |> Seq.filter(fun assem -> assem <> "Anonymously Hosted DynamicMethods Assembly")
+                            // RazorEngine generated
+                            |> Seq.filter(fun assem -> assem <> "RazorEngine.Compilation.ImpromptuInterfaceDynamicAssembly")
+                            |> Seq.cache
                         if dynamicAssemblies |> Seq.length > 0 then
-                            if printDetails then 
-                                trace "Dynamic assemblies were generated during evaluation of script.\nCan not save cache." 
+                            if printDetails then
```

Comment:
I think we should always show a message when caching was not possible instead of only with `--printdetails`. (Which means I think we should removing this line)